staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go

```diff
@@ -168,31 +171,39 @@ func (le *LeaderElector) IsLeader() bool {
 	return le.observedRecord.HolderIdentity == le.config.Lock.Identity()
 }
 
-// acquire loops calling tryAcquireOrRenew and returns immediately when tryAcquireOrRenew succeeds.
-func (le *LeaderElector) acquire() {
-	stop := make(chan struct{})
+// acquire loops calling tryAcquireOrRenew and returns true immediately when tryAcquireOrRenew succeeds.
+// Returns false if ctx signals done.
+func (le *LeaderElector) acquire(ctx context.Context) bool {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
```

Comment:
Why are you defering a cancel the cancel should only occur then they've lost leadership.

Reply:
This is a sub context, it does not influence the parent context. This code is just following the best practice patten where the resource is closed/freed when it falls out of scope.
Actually, just inspected how child cancel contexts are implemented - child contexts are referenced by parent contexts if both are cancel contexts. References are only cleaned up once the that `cancel()` function is called. So it is a good idea to clean things up.